File: SWEA/D3/1225. ［S／W 문제해결 기본］ 7일차 － 암호생성기/［S／W 문제해결 기본］ 7일차 － 암호생성기.java
import java.util.ArrayList;
import java.util.Scanner;

public class Solution {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        for(int i = 1; i <=10; i++){
            int n = scanner.nextInt();
            scanner.nextLine();

            ArrayList<Integer> data = new ArrayList<Integer>();
            String[] input = scanner.nextLine().split(" ");

            for(String s : input){
                data.add(Integer.parseInt(s));
            }

            int s = 0;
            for (;;) {
                int tmp = data.remove(0);
                tmp -= (s + 1);
                s = (s + 1) % 5;

                if (tmp <= 0) {
                    data.add(0);
                    break;
                }
                else{
                    data.add(tmp);
                }
            }
            System.out.print("#" + i +  " ");
            for(int k =0; k < data.size(); k++){
                System.out.print(data.get(k) + " ");
            }
            System.out.println();

        }
    }
}

File: SWEA/D3/1225. ［S／W 문제해결 기본］ 7일차 － 암호생성기/［S／W 문제해결 기본］ 7일차 － 암호생성기.py
"""
문제 이해
    - 입력:
        테스트 케이스의 번호 : 정수
        8개의 숫자를 입력 받음  : 정수 리스트
    - 출력:
        모든 사이클을 돌린 후 암호
아이디어:
    완전 탐색으로 진행
    일종의 큐?
    마지막 숫자가 0이 되면 종료
    01234
    i = 0
    while data[-1] != 0:
        tmp = data.pop(0)
        tmp -= (i+1)
        i = (i+1) % 5
        data.append(tmp)
    print(*data)
"""
for t in range(1, 11):
    n = int(input())
    data = list(map(int, input().split()))

    # 1 ~ 5 사이클을 돌기 위한 변수
    i = 0
    while True:
        # 헤드 pop
        tmp = data.pop(0)

        # 1 ~ 5 사이클을 돌며 -연산
        tmp -= (i + 1)

        # 사이클을 위한 인덱스 값 변화
        i = (i + 1) % 5

        # 만약 연산한 값이 0보다 작거나 같다면 반복문 중단
        if tmp <= 0:
            data.append(0)
            break
        
        # 0보다 크다면 계속해서 연산
        else:
            data.append(tmp)

    # 결과 출력
    print(f"#{t} ", end="")
    print(*data)
